activation/nipost.go

```diff
@@ -251,11 +263,19 @@ func (nb *NIPostBuilder) BuildNIPost(
 		return nil, fmt.Errorf("submitting to poets: %w", err)
 	}
 
-	// Phase 1: query PoET services for proofs
+	// Stage 1: query PoET services for proofs
+	if err := nb.identityStates.Set(signer.NodeID(), types.IdentityStateWaitForPoetRoundEnd); err != nil {
+		nb.logger.Warn("failed to switch identity state",
+			zap.Stringer("smesherID", signer.NodeID()),
+			zap.Error(err),
+		)
```

Comment:
Do we intend to continue in this case?

Reply:
Yes, because states - it's more 'internal' staff (imo), and it shouldn't break the process of registration to poet round

Reply:
On the other hand, failing to switch state means a serious logic bug, right? It might only fail if the state transition is not allowed so it could just crash as well.